### PR TITLE
Allow clients to control how much to give in GiftTransactions

### PR DESCRIPTION
**Pull request by E-Mac27**
# Summary
Clients previously was forced to give the amount they promised initially. This isn't quite inline with IITO design. Since giftsTransactions execute at the end of turn, clients state have changed.
  
Now they are able to specify if they want to fulfil the amount they initially promised or change it. 

## Test Plan
Oops.

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

**Review by NGovani — APPROVED** [review]

**Review by Eirikalb — APPROVED** [review]
LGTM, could you also remove the sendGift and receiveGift methods as these are obsolete as an effect of this?

File: internal/server/iito.go [hunk 1]
```diff
@@ -230,14 +230,19 @@ func (s *SOMASServer) distributeGiftHistory(totalResponses map[shared.ClientID]s
 func (s *SOMASServer) executeTransactions(transactions map[shared.ClientID]shared.GiftResponseDict) {
```
Comment (lhl2617) [inline internal/server/iito.go:230]: whomst'd've todo is this

**Comment by NGovani** [comment]
> LGTM, could you also remove the sendGift and receiveGift methods as these are obsolete as an effect of this?

They are still needed to notify the clients that their resources have changed. I believe they were changed in a previous PR to sentGift and recievedGift, just a change in tense

File: internal/server/iito.go (hunk 1 above)
Comment (E-Mac27) [reply internal/server/iito.go:230]: errrrrrrrrrrrrrrrrrrrrrrrrrr

**Pull request merged**